### PR TITLE
changes in playground command

### PR DESCRIPTION
**Pull request by udit-gulati**

### CODE REVIEW

File: packages/wasmkit/src/internal/cli/initialize-playground.ts [hunk 1]
```diff
@@ -57,28 +90,39 @@ async function checkDir (destination: string, force: boolean): Promise<void> {
...
-      name: 'Select an option',
-      message: 'Do you want to pick an existing template or exit?',
-      choices: ['Pick an existing template', 'exit']
+  // const templatePath = path.join(basePath, templateName);
+  const templatePath = path.join(basePath);
+  if (fse.existsSync(templatePath)) {
+    return [templatePath, templateName];
+  } else {
+    console.log(
+      chalk.red(
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/initialize-playground.ts:102]: Create a `WasmKitError` here and throw that.

File: packages/wasmkit/src/internal/cli/initialize-playground.ts [hunk 2]
```diff
@@ -57,28 +90,39 @@ async function checkDir (destination: string, force: boolean): Promise<void> {
...
+  templateName: string
 ): Promise<[string, string]> {
-  const templatePath = path.join(basePath, templateName);
-  if (fse.existsSync(templatePath)) { return [templatePath, templateName]; } else {
-    console.log(chalk.red(`Error occurred: template "${templateName}" does not exist in ${TEMPLATES_GIT_REMOTE}`));
-    const prompt = new (enquirer as any).Select({ // eslint-disable-line  @typescript-eslint/no-explicit-any
-      name: 'Select an option',
-      message: 'Do you want to pick an existing template or exit?',
-      choices: ['Pick an existing template', 'exit']
+  // const templatePath = path.join(basePath, templateName);
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/initialize-playground.ts:96]: Remove this comment

File: packages/wasmkit/src/internal/cli/playground-creation.ts [hunk 3]
```diff
@@ -1,47 +1,71 @@
...
     const filePath = path.join(contractDir, file);
     const yamlFile = fs.readFileSync(filePath, "utf8");
-    const yamlData = yaml.load(yamlFile) as CounterData;
-    const codeId = yamlData.default.deployInfo.codeId;
-    const codeAddress = yamlData.default.instantiateInfo[0].contractAddress;
-    const jsonData = {
-      [fileName]: {
-        codeId,
-        codeAddress
+    const yamlData = yaml.load(yamlFile) as Record<any, any>;
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/playground-creation.ts:40]: Record type should always be used as `Record<string, unknown>`, `any` is allow it as any type, `unknown` is DONT allow it as any type. That's why should use unknown most often in place of any, cause any hides type checking, and unknown forces type checking for those values that we don't yet know the type of. Hence, first should be the exact type, second unknown and then any.

File: packages/wasmkit/src/internal/cli/playground-creation.ts [hunk 4]
```diff
@@ -247,18 +267,35 @@ export function createConfirmationPrompt (
...
+      console.error(data.toString());
+    });
 
-//   return dep in allDependencies;
-// }
+    childProcess.on("close", (code: number) => {
+      if (code === 0) {
+        resolve(true);
+      } else {
+        reject(new Error(`Command failed with code: ${code}`));
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/playground-creation.ts:292]: WasmKitError

File: packages/wasmkit/src/internal/cli/playground-creation.ts [hunk 5]
```diff
@@ -247,18 +267,35 @@ export function createConfirmationPrompt (
...
+      }
+    });
 
-// function isYarnProject (): boolean {
-//   return fsExtra.pathExistsSync("yarn.lock");
-// }
+    childProcess.on("error", (error: ExecException) => {
+      console.error("Error executing command:", error);
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/playground-creation.ts:297]: WasmKitError

File: packages/wasmkit/src/internal/cli/initialize-playground.ts [hunk 6]
```diff
@@ -129,20 +180,48 @@ export async function initialize (
   const tempDirPath = tempDir.path;
   const tempDirCleanup = tempDir.cleanupCallback;
 
-  console.info(`* Fetching templates from ${TEMPLATES_GIT_REMOTE} *`);
+  console.info(
```
Comment (udit-gulati) [inline packages/wasmkit/src/internal/cli/initialize-playground.ts:183]: Fix the spacing here:

Should be:
```
Fetching templates from: https://github.com/arufa-research/wasmkit-react-playground
```

**Comment by udit-gulati** [comment]
@VarunProhit there should a way to update contract addresses and schema for playground app. 

For example, user creates a counter contracts, deploys to juno testnet, creates playground app, deploys to neutron testnet and now wants to see neutron contracts too in playground. Another could be deploy to juno testnet, create playground, upadte contract logic to add more query methods, compile the contracts again, deploy to juno testnet again, now the plaground should have the new query messages and new contract addresses in the react app.

This could be done as either of two approaches:

1. Allow user to update the plagound react app by a command `wasmkit playground-update` and the json config files for bot addresses and schema will be updated in the react app.
2. Wrap the `yarn start` of the playground/ app inside a `wasmkit playground-start` which does yarn start at last but first checks if the contract addr and schema json files in react app are correct (config validation, this is useful to identify if any manual change in this cfg by the user is causing any issues), then it should check if this json cfg matches the artifacts, if not then prompt user to y/n on overwriting the json with new values from artifacts, then finally start the react app.

I think the best approach here is to do both, so that user has more information on what is happening while they are starting the playground or creating it, plus it'll simplify the process too and align it with wasmkit commands more.

File: packages/wasmkit/src/internal/cli/playground-creation.ts (hunk 3 above)
Comment (VarunProhit) [reply packages/wasmkit/src/internal/cli/playground-creation.ts:40]: Record<string, CheckpointInfo>   i have made type of CheckpointInfo and used it that has resolved issue.

**Pull request merged**